Framework/HistogramData/test/RebinTest.h

```diff
@@ -0,0 +1,446 @@
+
+using namespace Mantid::HistogramData;
+
+class RebinTest : public CxxTest::TestSuite {
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static RebinTest *createSuite() { return new RebinTest(); }
+  static void destroySuite(RebinTest *suite) { delete suite; }
+
+  void testExecrebin() {
+    TS_ASSERT_THROWS_NOTHING(
+        rebin(getCountsHistogram(), BinEdges(10, LinearGenerator(0, 0.5))));
+  }
+
+  void testExecRebinFrequency() {
+    TS_ASSERT_THROWS_NOTHING(
+        rebin(getFrequencyHistogram(), BinEdges(10, LinearGenerator(0, 0.5))));
+  }
+
+  void testRebinNoYModeDefined() {
+    BinEdges edges(5, LinearGenerator(0, 2));
+    // X Mode Points
+    TS_ASSERT_THROWS(
+        rebin(Histogram(Histogram::XMode::Points, Histogram::YMode::Counts),
+              edges),
+        std::runtime_error);
+    // No YMode set
+    TS_ASSERT_THROWS(
+        rebin(Histogram(BinEdges(10, LinearGenerator(0, 0.5))), edges),
+        std::runtime_error);
+  }
+
+  void testRebinFailsCentralBinEdgesInvalid() {
+    std::vector<double> binEdges{1, 2, 3, 3, 5, 7};
+    BinEdges edges(binEdges);
+
+    TS_ASSERT_THROWS(rebin(getCountsHistogram(), edges), std::runtime_error);
+  }
+
+  void testRebinFailsStartBinEdgesInvalid() {
+    std::vector<double> binEdges{1, 1, 3, 4, 5, 7};
+    BinEdges edges(binEdges);
+
+    TS_ASSERT_THROWS(rebin(getCountsHistogram(), edges), std::runtime_error);
+  }
+
+  void testRebinEndCentralBinEdgesInvalid() {
+    std::vector<double> binEdges{1, 2, 3, 4, 5, 5};
+    BinEdges edges(binEdges);
+
+    TS_ASSERT_THROWS(rebin(getCountsHistogram(), edges), std::runtime_error);
+  }
+
+  void testRebinFailsInputBinEdgesInvalid() {
+    std::vector<double> binEdges{1, 2, 3, 3, 5, 7};
+    Histogram hist(BinEdges(std::move(binEdges)), Counts(5, 10));
+    BinEdges edges{1, 2, 3, 4, 5, 6};
+
+    TS_ASSERT_THROWS(rebin(hist, edges), std::runtime_error);
+  }
+
+  void testRebinFailsInputAndOutputBinEdgesInvalid() {
+    std::vector<double> binEdges{1, 2, 3, 3, 5, 7};
+    Histogram hist(BinEdges(binEdges), Counts(5, 10));
+    BinEdges edges(std::move(binEdges));
+
+    TS_ASSERT_THROWS(rebin(hist, edges), std::runtime_error);
+  }
+
+  void testRebinIdenticalBins() {
+    auto histCounts = getCountsHistogram();
+    auto histFreq = getFrequencyHistogram();
+
+    auto outCounts = rebin(histCounts, histCounts.binEdges());
+    auto outFreq = rebin(histFreq, histFreq.binEdges());
+
+    TS_ASSERT_EQUALS(outCounts.x().rawData(), histCounts.x().rawData());
+    TS_ASSERT_EQUALS(outCounts.y().rawData(), histCounts.y().rawData());
+    TS_ASSERT_EQUALS(outCounts.e().rawData(), histCounts.e().rawData());
+
+    TS_ASSERT_EQUALS(outFreq.x().rawData(), histFreq.x().rawData());
+    TS_ASSERT_EQUALS(outFreq.y().rawData(), histFreq.y().rawData());
+    TS_ASSERT_EQUALS(outFreq.e().rawData(), histFreq.e().rawData());
```

Comment:
Didn't you once write that test helper that gives `operator==` so you don't have to write `rawData` every time?